secrets/scaleway/src/main/kotlin/ScalewayConfiguration.kt

```diff
@@ -34,6 +39,31 @@ data class ScalewayConfiguration(
 
         val DEFAULT_API_VERSION = ScalewayApiVersion.V1_BETA1
         val DEFAULT_REGION = ScalewayRegion.FRANCE_PARIS
+
+        fun create(configManager: ConfigManager): ScalewayConfiguration =
+            with(serializer().descriptor) {
+                val apiVersionName = getElementName(1)
```

Comment:
What's the benefit of referencing the properties by their index instead of name? It looks like you are just replacing hard-coded names with magic numbers.

Reply:
In a way you're right, but at least this way, if a property name changes, the hard-coded string does not need to be renamed as well. I saw all the repetition via constants in the `VaultConfiguration` and didn't like it, so I was looking for a different approach.

Reply:
But the code looks more cryptic and if the property order changes you would have the same issue which might be even harder to find. In both cases there should be tests that prevent errors when properties change.

> I saw all the repetition via constants in the VaultConfiguration and didn't like it,

I don't like that either, maybe we could think about a generic approach that uses reflection and does not depend on kotlinx.serialization in future.

Anyway, as this code is isolated and it's urgent for you I don't want to block this here.

Reply:
> I don't like that either, maybe we could think about a generic approach that uses reflection and does not depend on kotlinx.serialization in future.

Avoid reflection (which would not work multi-platform) by leveraging kotlinx.serialization was exactly the charming aspect of it. But I agree that a generic solution built into ConfigManager or so would be nicer.

Reply:
> I you prefer hard-coding the property names, I can do that quickly.

Done.